modules/asg_node_group/versions.tf

```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 3.49"
+      version = "~> 4.16"
```

Comment:
I think there's going to be a deployment challenge here as the old AWS provider constraint was `~> 3.49` and the new one is `~>4.17`. These two constraints are incompatible so we won't be able to upgrade some clusters to this module version and not others within a single project? 😭

Reply:
Might we be able to change the older 1.21-rc module to `>=3.49` so there's at least a pathway? That seems risky too, though, as that's allowing future major version bumps even beyond 4. 😭

Reply:
we can add an upper constraint:

Lets do this:

* for 1.22 change the constraint to: `>= 3.49.0, < 5.0.0`
* Add something to release notes saying that for 1.22 we support provider versions 3 and 4 (but 4 is recommended) and that 1.23 will drop support for 3.
* We will have to be careful not to use any features only available in provider v4 in the 1.22 release and should do some manual validation of the rc against both versions of the provider - but keep the automated tests running against v4.

Reply:
@jportasa - this might be a big / tricky enough topic that it makes sense to extract this provider version stuff to a separate PR - as this would make it easier to cherry-pick a commit onto 1.21 if we decide that that is needed?

Reply:
fixed